### PR TITLE
Input validation for selection data containers 

### PR DESCRIPTION
**Pull request by git-abhishek**
## PR Summary
Fix for #1843.

## PR Checklist

- [X] Code passes flake8 checker
- [ ] New features are documented, with docstrings and narrative docs **NA**
- [X] Adds a test for any bugs fixed. Adds tests for new features.

## Adding Reviewers
@ngoldbaum @Xarthisius @colinmarc

### CODE REVIEW

File: yt/funcs.py [hunk 1]
```diff
@@ -1231,3 +1231,29 @@ def validate_object(obj, data_type):
         raise TypeError("Expected an object of '%s' type, received '%s'"
                         % (str(data_type).split("'")[1],
                         str(type(obj)).split("'")[1]))
+
+def validate_axis(axis):
+    err_msg = ("Expected axis of int or char type ( can be "
+               "[0, 1, 2, '0', '1', '2', 'x', 'y', 'z']), received '%s'.")
```
Comment (ngoldbaum) [inline yt/funcs.py]: This won't work for non-cartesian data. You probably want to populate this list from e.g. `ds.coordinates.axis_.id.keys()`.

Comment (git-abhishek) [reply yt/funcs.py]: Thanks, fixed.

**Comment by ngoldbaum** [comment]
Looks like the tests are still failing here.

**Comment by git-abhishek** [comment]
The error is due to string different error message in assert statement. I  was comparing a list of string, so will sort it and then compare. Will fix it shortly.

**Pull request merged**